client-java/src/main/java/uk/gov/gchq/palisade/client/Resource.java

```diff
@@ -13,54 +13,43 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package uk.gov.gchq.palisade.client.job.state;
+package uk.gov.gchq.palisade.client;
+
+import java.util.Optional;
 
 /**
- * The status of a download
+ * A resource message
  *
  * @since 0.5.0
  */
-public enum JobStatus {
-
-    /**
-     * Completed successfully
-     */
-    OPEN(10),
-
-    /**
-     * A request has been sent, but not received
-     */
-    REQUEST_SENT(15),
+public interface Resource extends Message {
 
     /**
-     * Download failed
+     * Returns the leaf resource id which is to be downloaded
+     *
+     * @return the leaf resource id which is to be downloaded
      */
-    RESPONSE_RECEIVED(20),
+    String getLeafResourceId();
 
     /**
-     * Downloads are in progress
+     * Return the url of the download service
+     *
+     * @return the url of the download service
```

Comment:
On reflection, the url should not be exposed to the user of the client. Once this is added to the API it could never be removed without breaking binary compatibility. I've removed it.

Reply:
I've changed it to this : "the URL location of the resource"